examples/step-56/doc/intro.dox

```diff
@@ -0,0 +1,159 @@
+<i>This program was contributed by Ryan Grove and Timo Heister.</i>
+
+<a name="Intro"></a> 
+<h1>Introduction</h1>
+
+<h3> Stokes Problem </h3>
+
+The purpose of this tutorial is to create an efficient linear solver
+for the Stokes equation and compare it to alternative
+approaches. Using FGMRES with geometric multigrid as a preconditioner
+for the velocity block, we see that the linear solvers used in step-22
+cannot keep up since multigrid is the only way to get $O(n)$ solve
+time. Using the Timer class, we collect some statistics to compare
+setup times, solve times, and number of iterations. We also compute
+errors to make sure that what we have implemented is correct.
+
+Let $u \in H_0^1 = \{ u \in H^1(\Omega), u|_{\partial \Omega} = 0 \}$
+and $p \in L_*^2 = \{ p \in L^2(\Omega), \int_\Omega p = 0
+\}$. The Stokes equations read as follows in non-dimensionalized form:
+
+@f{eqnarray*}
+ - 2 \text{div} \frac {1}{2} \left[ (\nabla \textbf{u})
+ + (\nabla \textbf{u})^T\right] + \nabla p & =& f \\
+ - \nabla \cdot u &=& 0 
+@f}
+
+Note that we are using the deformation tensor instead of $\Delta u$ (a
+detailed description of the difference between the two can be found in
+step-22, but in summary, the deformation tensor is more physical as
+well as more expensive).
+
+<h3> Linear Solver and Preconditioning Issues </h3>
+
+The weak form of
+the discrete equations naturally leads to the following linear system
+for the nodal values of the velocity and pressure fields:
+@f{eqnarray*} 
+\left(\begin{array}{cc} A & B^T \\ B & 0
+\end{array}\right) \left(\begin{array}{c} U \\ P \end{array}\right) =
+\left(\begin{array}{c} F \\ 0 \end{array}\right). 
+@f}
+
+Our goal is to compare several solution approaches.  While step-22
+solves the linear system using a "Schur complement approach" in two
+separate steps, we instead attack the
+block system at once using FMGRES with an efficient
+preconditioner. The idea is as follows: if we find a block
```

Comment:
-> ...with an efficient preconditioner, in the spirit of the approach outlined in the "Results" section of step-22.